Source/WebCore/rendering/mathml/RenderMathMLToken.cpp

```diff
@@ -573,10 +573,7 @@ void RenderMathMLToken::layoutBlock(bool relayoutChildren, LayoutUnit pageLogica
 {
     ASSERT(needsLayout());
 
-    for (auto& box : childrenOfType<RenderBox>(*this)) {
-        if (box.isOutOfFlowPositioned())
-            box.containingBlock()->insertPositionedObject(box);
-    }
+    handlePositionedAndFloatingChildren();
```

Comment:
Other methods do that after the "!relayoutChildren && simplifiedLayout()" early return. Have you checked what non-MathML classes do?

https://searchfox.org/wubkat/search?q=%21relayoutChildren+%26%26+simplifiedLayout%28%29&path=Source%2FWebCore%2Frendering&case=false&regexp=false

It seems at least for flexbox and grid, the insertPositionedObject calls are done *after* the early return.

Reply:
Since this was the previous behaviour will this change anything? I don't think so (mimics what is done for other elements, so it should be fine) but want to make sure

Reply:
Good question. I believe the only difference is that  the OOF children are inserted yet in the containing block when `simplifiedLayout()` is executed. But we don't care since it's up to the containing block (which is an ancestor) to take care of the layout of these children. `simplifiedLayout()` calls `layoutPositionedObjects()` for the objects from its own positioned objects map (which were inserted by some descendants).

For this particular case, I guess you checked the test added in https://commits.webkit.org/262449@main does not crash in debug?

Reply:
mmh, actually the containingBlock() will likely return the mathml renderer itself rather than an ancestor, we will need to think about it.

Reply:
For future reference, Harry and I just had a meeting where we talked about it.

Probably we need to do the insertion of positioned children before the `simplifiedLayout()`, especially for MathML where `child->containingBlock()` is the object itself. Otherwise https://searchfox.org/wubkat/rev/b36cbce69fddb7da33823f316bd8ead5bebee970/Source/WebCore/rendering/RenderBlock.cpp#847 won't layout the inserted positioned children and nobody else we do it for us. (Perhaps for grid/flex, the containing block is always an ancestor so we don't have this issue).

For this edge case to happen, a test case needs to insert a positioned absolute object and do some extra changes that would trigger a `simplifiedLayout()`. Not sure whether that's already cover and how easy it would be to write one, though.

Doing the layout of floats before `simplifiedLayout()` could have a perf impact, so probably better to continue doing it after the `simplifiedLayout()` call, and so split `handlePositionedAndFloatingChildren()`. That's a pity to have multiple loops over the children, but hopefully won't have big impact on performance because each loop is skipping  "irrelevant" children.